### PR TITLE
Abdm facility

### PR DESCRIPTION
**Pull request by helenKaryamsetty**
## 📋 Description

JIRA ID: AMM-894

Please provide a summary of the change and the motivation behind it. Include relevant context and details.

---

## ✅ Type of Change

- [ ] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [ ] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new REST API endpoint for retrieving ABDM facility details based on work location.
	- Added new fields for ABDM facility ID and name to enhance data mapping.
	- Created a repository for managing ABDM facility data with custom query capabilities.
	- Established a service interface and implementation for handling ABDM facility operations.

- **Bug Fixes**
	- Improved error handling and response structure for ABDM facility requests.

### CODE REVIEW

**Review by ravishanigarapu — APPROVED** [review]
looks fine

File: src/main/java/com/iemr/common/service/abdmfacility/AbdmFacilityService.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.iemr.common.service.abdmfacility;
+
+public interface AbdmFacilityService {
+
+	public String getMappedAbdmFacility(int psmId);
+
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/iemr/common/service/abdmfacility/AbdmFacilityService.java:6]: @coderabbitai i just need single value from that.. Creating a DTO for it is unnecessary so directly using String

File: src/main/java/com/iemr/common/repository/abdmfacility/AbdmFacilityRepository.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import com.iemr.common.data.users.ProviderServiceAddressMapping;
+
+@Repository
+public interface AbdmFacilityRepository extends CrudRepository<ProviderServiceAddressMapping, Integer> {
+	
+	@Query("SELECT v from ProviderServiceAddressMapping v where v.pSAddMapID = :pssmID")
+	ProviderServiceAddressMapping getAbdmFacility(@Param("pssmID") int pssmID);
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/iemr/common/repository/abdmfacility/AbdmFacilityRepository.java:14]: @coderabbit That response contains only single object of list. Oder By is not required

File: src/main/java/com/iemr/common/service/abdmfacility/AbdmFacilityServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	
+	@Autowired
+	private AbdmFacilityRepository abdmFacilityRepo;
+	
+	@Override
+	public String getMappedAbdmFacility(int psmId) {
+		ProviderServiceAddressMapping res = abdmFacilityRepo.getAbdmFacility(psmId);
+		return res.toString();
+		
+	}
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/iemr/common/service/abdmfacility/AbdmFacilityServiceImpl.java:20]: @coderabbitai i want to return exact response. dont want to throw any error whatever it is... so not throwing any expections. Its handled in UI

**Pull request merged**